ddtrace/tracer/spancontext.go

```diff
@@ -84,6 +87,23 @@ func (c *spanContext) SpanID() uint64 { return c.spanID }
 // TraceID implements ddtrace.SpanContext.
 func (c *spanContext) TraceID() uint64 { return c.traceID }
 
+// TraceID128 implements ddtrace.SpanContextW3C.
+func (c *spanContext) TraceID128() string {
+	hiBits := c.span.Meta[keyTraceID128]
+	if hiBits == "" {
+		return "" // 128 bit trace ids not enabled
+	}
+	hi, err := hex.DecodeString(hiBits)
+	if err != nil {
+		return ""
+	}
+	buf := make([]byte, 16)
+	copy(buf[:8], hi)
+	binary.BigEndian.PutUint64(buf[8:], c.traceID)
+	id := hex.EncodeToString(buf)
+	return strings.TrimLeft(id, "0") // trim any leading zeros
```

Comment:
I'm wondering if we should leave the zeros in.

At least for the W3C headers, the zeros appear to be required, and the ID string is expected to be exactly 32 characters. Depending on where this string is used, trimming the zeros may result in parsing problems.

Reply:
Yep I agree with you. I was debating this, but actually leaving the zeros in will make for better consistency across the board. I decided to also return leading zeros as needed for the `SpanContextW3C.TraceID128` function as well 👍